### ISSUE TITLE
Not working at all [OSX]

### ISSUE THREAD

**Issue by sarink** [AtomLinter/linter-tslint#221]
plugin version: 1.9.0

atom version: 1.25.0

tslint.json
```
{
  "extends": "tslint-config-airbnb"
}
```

```
:$ npm get prefix
/Users/sarink/.nvm/versions/node/v8.7.0

:$ ls node_modules | grep tslint
tslint
tslint-config-airbnb
tslint-consistent-codestyle
tslint-eslint-rules
tslint-microsoft-contrib

:$ cat package.json | grep tslint
    "tslint": "^5.9.1",
    "tslint-config-airbnb": "^5.4.2",
```

<img width="754" alt="screen shot 2018-03-26 at 4 57 55 pm" src="https://user-images.githubusercontent.com/1881761/37935424-11d8afac-3117-11e8-9fac-945d42ce9247.png">

Produces no warnings, it's not even clear if the linter is running at all. Everything was working fine until a week or two ago?

Running tslint manually from the command line works fine.

Any ideas?

**Comment by Xapphire13** [comment]
Can you open the atom console (<kbd>CMD</kbd>+<kbd>OPT</kbd>+<kbd>I</kbd> on Mac), open a file that should have lint issues, and see if anything shows up there?

Make sure the log level is `verbose` just to be safe:

![image](https://user-images.githubusercontent.com/3392349/37938445-bd7d583a-3112-11e8-8dde-f2b005f16ff0.png)

**Comment by sarink** [comment]
```
/Users/sarink/.atom/packages/linter-tslint/lib/workerHelper.js:46 Uncaught (in promise) TypeError: Cannot read property 'on' of undefined
    at /Users/sarink/.atom/packages/linter-tslint/lib/workerHelper.js:34:34
    at Promise (<anonymous>)
    at Object.requestJob (/Users/sarink/.atom/packages/linter-tslint/lib/workerHelper.js:33:10)
    at /Users/sarink/.atom/packages/linter-tslint/lib/main.js:83:34
    at Generator.next (<anonymous>)
    at step (/Users/sarink/.atom/packages/linter-tslint/lib/main.js:9:273)
    at /Users/sarink/.atom/packages/linter-tslint/lib/main.js:9:443
    at Promise (<anonymous>)
    at /Users/sarink/.atom/packages/linter-tslint/lib/main.js:9:99
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:141:34)
    at /Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/text-buffer.js:1359:31
    at <anonymous>
```

**Comment by Xapphire13** [comment]
Interesting, looks like an edge case, your editor is triggering the [lint on save](https://github.com/AtomLinter/linter-tslint/blob/master/lib/main.js#L82-L84) before the [worker is created](https://github.com/AtomLinter/linter-tslint/blob/cd77270eeacba1e38f95b079cd863232d4dabfef/lib/main.js#L148-L152).

The expectation would be that the `provideLinter()` method would be called before a save was possible. Seems this isn't the case and the `if (!this.worker)` code should be added to the `lintOnSave`

**Comment by nrkroeker** [comment]
Is there a way to get this fix? I have the exact same error, and I know it was only resolved within the last couple weeks but since there isn't a new release, would I need to clone this repository instead of using the apm version?

**Comment by sarink** [comment]
In the meantime, I wrote a custom lint watcher out of tslint and chokidar. Here it is, if anyone's interested (added to `package.json`):
```
    "lint": "tslint --project .",
    "lint:watch": "SHELL=/bin/bash chokidar tslint.json ./**/*.ts ./**/*.tsx -c \"npm run lint\" --initial --verbose",
```

**Comment by nrkroeker** [comment]
@sarink I’m unfamiliar with lint watchers, does it only affect tslint’s functionality when run from the command line? Or in atom as well?

**Comment by Xapphire13** [comment]
Sorry, didn't get around to making a release. I will get onto that now

**Comment by Xapphire13** [comment]
Just published version 1.9.1. Please try with that version
